src/libs/Violations/ViolationsUtils.ts

```diff
@@ -78,6 +79,53 @@ const ViolationsUtils = {
                 if (!hasMissingTagViolation && !updatedTransaction.tag && policyRequiresTags) {
                     newTransactionViolations.push({name: CONST.VIOLATIONS.MISSING_TAG, type: 'violation'});
                 }
+            } else {
+                newTransactionViolations = newTransactionViolations.filter(
+                    (violation) => violation.name !== CONST.VIOLATIONS.SOME_TAG_LEVELS_REQUIRED && violation.name !== CONST.VIOLATIONS.TAG_OUT_OF_POLICY,
+                );
+
+                // We first get the errorIndexes for someTagLevelsRequired. If it's not empty, we puth SOME_TAG_LEVELS_REQUIRED in Onyx.
+                // Otherwise, we put TAG_OUT_OF_POLICY in Onyx (when applicable)
+                const errorIndexes = [];
+                for (let i = 0; i < policyTagKeys.length; i++) {
+                    const isTagRequired = policyTagList[policyTagKeys[i]].required ?? true;
+                    const isTagSelected = Boolean(selectedTags[i]);
+                    if (isTagRequired && (!isTagSelected || (selectedTags.length === 1 && selectedTags[0] === ''))) {
+                        errorIndexes.push(i);
+                    }
+                }
+                if (errorIndexes.length !== 0) {
```

Comment:
NAB: since 0 is falsy in JS this can be simplified.
```suggestion
                if (errorIndexes.length) {
```